academy/3-ibc/6-ica.md

```diff
@@ -16,635 +15,297 @@ tags:
+Therefore, the provided API on the host submodule consists of:
 
-The **authentication module** must:
+* `RegisterInterchainAccount`: enables the registration of Interchain accounts on the host, associated with an owner on the controller side.
+* `ExecuteTx`: enables the transaction data to be executed, provided successful authentication.
+* `AuthenticateTx`: checks that the signer of a particular message is the Interchain account associated with the counterparty portID of the channel that the IBC packet was sent on.
 
-* Authenticate interchain account owners
-* Track the associated interchain account address for an owner
-* Claim the channel capability in `OnChanOpenInit`
-* Send packets on behalf of an owner (after authentication)
+<HighlightBox type="note">
 
-The following [`IBCModule`](https://github.com/cosmos/ibc-go/blob/main/modules/core/05-port/types/module.go) callbacks must be implemented with appropriate custom logic:
+The host API methods run automatically as part of the flow and need not be exposed to an end-user or module, as is the case on the controller side with `RegisterInterchainAccount` and `SendTx`.
 
-```go
-// OnChanOpenInit implements the IBCModule interface
-func (im IBCModule) OnChanOpenInit(
-    ctx sdk.Context,
-    order channeltypes.Order,
-    connectionHops []string,
-    portID string,
-    channelID string,
-    chanCap *capabilitytypes.Capability,
-    counterparty channeltypes.Counterparty,
-    version string,
-) error {
-    // the authentication module *must* claim the channel capability on OnChanOpenInit
-    if err := im.keeper.ClaimCapability(ctx, chanCap, host.ChannelCapabilityPath(portID, channelID)); err != nil {
-        return err
-    }
-
-    // perform custom logic
-
-    return nil
-}
+</HighlightBox>
 
-// OnChanOpenAck implements the IBCModule interface
-func (im IBCModule) OnChanOpenAck(
-    ctx sdk.Context,
-    portID,
-    channelID string,
-    counterpartyVersion string,
-) error {
-    // perform custom logic
+#### Register an Interchain account
 
-    return nil
-}
+The `RegisterInterchainAccount` flow was discussed on the controller side already, where it triggered a handshake. On the host side is a complementary part of the flow, but here it's triggered in the `OnChanOpenTry` step of the handshake, which will create the Interchain account.
```

Comment:
Does anyone see if I forgot some formatting here somewhere? The syntax highlighting seems messed up from here on forward but I cannot identify why, maybe a fresh pair of eyes will see...